lib/Jobs/MigrateGroups.php

```diff
@@ -97,20 +110,30 @@ protected function migrateGroup(string $gid): bool {
 			if ($affected === 0) {
 				throw new \RuntimeException('Could not delete group from local backend');
 			}
-
 			if (!$this->ownGroupBackend->createGroup($gid)) {
 				throw new \RuntimeException('Could not create group in SAML backend');
 			}
 
 			$this->dbc->commit();
-
-			return true;
-		} catch (\Exception $e) {
+			$isMigrated = true;
+		} catch (Throwable $e) {
 			$this->dbc->rollBack();
 			$this->logger->warning($e->getMessage(), ['app' => 'user_saml', 'exception' => $e]);
 		}
 
-		return false;
+		if ($allUsersInserted && $isMigrated) {
+			try {
+				$this->groupMigration->cleanUpOldGroupUsers($gid);
+			} catch (Exception $e) {
+				$this->logger->warning('Error while cleaning up group members in (oc_)group_user of group (gid) {gid}', [
+					'app' => 'user_saml',
```

Comment:
Isn't the app already set when getting the LoggerInterface through DI?

Reply:
> Isn't the app already set when getting the LoggerInterface through DI?

I am never sure of that :see_no_evil:  Sometimes you see logs without an app set.